File: src/main/java/indi/goddess/shoppingmall2/service/manager/BrandService.java
package indi.goddess.shoppingmall2.service.manager;

import indi.goddess.shoppingmall2.beans.TbBrand;
import indi.goddess.shoppingmall2.beans.group.Specification;
import indi.goddess.shoppingmall2.entity.PageResult;

import java.util.List;
import java.util.Map;

//品牌列表
public interface BrandService {
    //返回所有品牌列表
    public List<TbBrand> findAllBrand();

    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加品牌
     */
    public void add(TbBrand brand);

    /**
     * 修改品牌
     */
    public void update(TbBrand brand);

    //根据id获取实体
    public TbBrand findOne(Long id);

    /**
     * 批量删除
     * @param ids
     */
    public void delete(Long[] ids);

    //条件查询
   public PageResult findPageByCondition(TbBrand tbBrand, int pageNum, int pageSize);

   //获取下拉列表
   List<Map> selectOptionList();


}

File: src/main/resources/static/manager/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
	this.findAll = function(){
		return $http.get("../brand/findAllBrand");
	}
	
	this.findPage = function(page,rows){
		return $http.get("../brand/findByPage?page="+page+"&rows="+rows);
	}
	
	this.save = function(entity){
		return $http.post("../brand/add",entity);
	}
	
	this.update=function(entity){
		return $http.post("../brand/update",entity);
	}
	
	this.findById=function(id){
		return $http.get("../brand/findOne?id="+id);
	}
	
	this.dele = function(ids){
		return $http.get("../brand/delete?ids="+ids);
	}

    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../brand/search?page='+page+"&rows="+rows, searchEntity);
    }

    this.selectOptionList = function(){
		return $http.get("../brand/selectOptionList");
	}
});